types/vote.go

```diff
@@ -109,6 +109,37 @@ func (vote *Vote) CommitSig() CommitSig {
 	}
 }
 
+// ExtendedCommitSig attempts to construct an ExtendedCommitSig from this vote.
+// Panics if either the vote extension signature is missing or if the block ID
+// is not either empty or complete.
+func (vote *Vote) ExtendedCommitSig() ExtendedCommitSig {
```

Comment:
I'm not sure that having this return a non-pointer is really getting us all that much.

Reply:
Again, monkey-see-monkey-do. This got "inspiration" from `CommitSig` function defined in line 89 above